modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1738,80 +1818,108 @@ public void testUpdateSystemObjectDefinition() throws Exception {
+				LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie", null,
+				null, false, LocalizedMapUtil.getLocalizedMap("Charlies"),
+				ObjectDefinitionConstants.SCOPE_SITE));
 
 		// Label is null
 
-		_assertFailure(
+		AssertUtils.assertFailure(
 			ObjectDefinitionLabelException.class,
 			"Label is null for locale " + LocaleUtil.US.getDisplayName(),
-			objectDefinition,
-			objectDefinitionId ->
-				_objectDefinitionLocalService.updateCustomObjectDefinition(
-					null, objectDefinitionId, 0, 0, 0, 0, false, false, false,
-					true, false, false, null, "Charlie", null, null, false,
-					LocalizedMapUtil.getLocalizedMap("Charlie"),
-					ObjectDefinitionConstants.SCOPE_SITE));
+			() -> _objectDefinitionLocalService.updateCustomObjectDefinition(
+				null, objectDefinition1.getObjectDefinitionId(), 0, 0, 0, 0,
+				false, false, false, true, false, false, null, "Charlie", null,
+				null, false, LocalizedMapUtil.getLocalizedMap("Charlie"),
+				ObjectDefinitionConstants.SCOPE_SITE));
 
 		// Plural label is null
 
-		_assertFailure(
+		AssertUtils.assertFailure(
 			ObjectDefinitionPluralLabelException.class,
 			"Plural label is null for locale " + LocaleUtil.US.getDisplayName(),
-			objectDefinition,
-			objectDefinitionId ->
-				_objectDefinitionLocalService.updateCustomObjectDefinition(
-					null, objectDefinitionId, 0, 0, 0, 0, false, false, false,
-					true, false, false,
-					LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie",
-					null, null, false, null,
-					ObjectDefinitionConstants.SCOPE_SITE));
+			() -> _objectDefinitionLocalService.updateCustomObjectDefinition(
+				null, objectDefinition1.getObjectDefinitionId(), 0, 0, 0, 0,
+				false, false, false, true, false, false,
+				LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie", null,
+				null, false, null, ObjectDefinitionConstants.SCOPE_SITE));
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition1);
 
 		// After Update a modifiable system object definition check its
 		// properties
 
-		objectDefinition =
+		ObjectDefinition objectDefinition2 =
+			ObjectDefinitionTestUtil.addModifiableSystemObjectDefinition(
+				TestPropsValues.getUserId(), null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Test", null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_SITE, null, 1,
+				_objectDefinitionLocalService,
+				Collections.singletonList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING,
+						RandomTestUtil.randomString(), StringUtil.randomId())));
+
+		objectDefinition2 =
 			_objectDefinitionLocalService.publishSystemObjectDefinition(
 				TestPropsValues.getUserId(),
-				objectDefinition.getObjectDefinitionId());
+				objectDefinition2.getObjectDefinitionId());
 
-		objectDefinition =
+		ObjectFolder objectFolder = _addObjectFolder();
```

Comment:
We need to delete this ObjectFolder at the end of this test.
Verify each use of this private method and check if we are deleting the ObjectFolder, please :)

Reply:
Done